pkg/skaffold/build/docker/docker.go

```diff
@@ -117,7 +117,7 @@ func (b *Builder) dockerCLIBuild(ctx context.Context, out io.Writer, workspace s
 		} else {
 			cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=0")
 		}
-	} else if matcher.IsNotEmpty() { // cross-platform builds require buildkit
+	} else if len(matcher.Platforms) == 1 { // cross-platform builds require buildkit
 		cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=1")
```

Comment:
Can you add a debug statement here indicating we set `Docker_Buildkit=1` and why.